builder/proxmox/common/step_start_vm.go

```diff
@@ -280,17 +265,73 @@ func generateProxmoxNetworkAdapters(nics []NICConfig) proxmox.QemuDevices {
 	return devs
 }
 
-func generateProxmoxDisks(disks []diskConfig) *proxmox.QemuStorages {
+func generateProxmoxDisks(disks []diskConfig, additionalISOFiles []additionalISOsConfig, bootiso string) *proxmox.QemuStorages {
 	ideDisks := proxmox.QemuIdeDisks{}
 	sataDisks := proxmox.QemuSataDisks{}
 	scsiDisks := proxmox.QemuScsiDisks{}
 	virtIODisks := proxmox.QemuVirtIODisks{}
 
+	// additionalISOsConfig accepts a static device type and index value in Device.
+	// Disks accept a device type but no index value.
+	//
+	// If this is a proxmox-iso build, the boot iso device is mapped after this function (builder/proxmox/iso/builder.go func Create)
+	// Map Additional ISO files first to ensure they get their assigned device index then proceed with disks in remaining available fields.
+	if len(additionalISOFiles) > 0 {
+		for _, iso := range additionalISOFiles {
+			// IsoFile struct parses the ISO File and Storage Pool as separate fields.
+			isoFile := strings.Split(iso.ISOFile, ":iso/")
+
+			// define QemuCdRom containing isoFile properties
+			bootIso := &proxmox.QemuCdRom{
+				Iso: &proxmox.IsoFile{
+					File:    isoFile[1],
+					Storage: isoFile[0],
+				},
+			}
+
+			// extract device type from ISODevice config value eg. ide from ide2
+			// validation of the iso.Device value occurs in builder/proxmox/common/config.go func Prepare
+			rd := regexp.MustCompile(`\D+`)
+			device := rd.FindString(iso.Device)
+			// extract device index from ISODevice config value eg. 2 from ide2
+			rb := regexp.MustCompile(`\d+`)
+			index, _ := strconv.Atoi(rb.FindString(iso.Device))
+
+			log.Printf("Mapping Additional ISO to %s%d", device, index)
+			switch device {
+			case "ide":
+				dev := proxmox.QemuIdeStorage{
+					CdRom: bootIso,
+				}
+				reflect.
```

Comment:
I wonder given this is very similar to what you added in an earlier PR, could we factorise this code that adds the device to the VM spec?